src/durableorchestrationclient.ts

```diff
@@ -294,45 +285,60 @@ export class DurableOrchestrationClient {
         createdTimeFrom: Date,
         createdTimeTo?: Date,
         runtimeStatus?: OrchestrationRuntimeStatus[],
-        ): Promise<PurgeHistoryResult> {
-        const idPlaceholder = this.clientData.managementUrls.id;
-        let requestUrl = this.clientData.managementUrls.statusQueryGetUri
-        .replace(idPlaceholder, "");
+    ): Promise<PurgeHistoryResult> {
+        let requestUrl: string;
+        if (this.clientData.rpcBaseUrl) {
+            // Fast local RPC path
+            let path = new URL("instances/", this.clientData.rpcBaseUrl).href;
+            const query: string[] = [];
+            if (createdTimeFrom) { query.push(`createdTimeFrom=${createdTimeFrom.toISOString()}`); }
```

Comment:
The main reason I added this new logic is because it's simpler, more efficient, and more readable. I didn't want to change the existing logic because it is legacy, should get removed eventually, and I didn't want to risk any regressions.